hooks/useCopyToClipboard.ts

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from 'react';
+
+const copyToClipboard = (value: string) => {
+  if (typeof window === 'undefined') {
+    return Promise.resolve(false);
+  }
+
+  return navigator.clipboard
+    .writeText(value)
+    .then(() => true)
+    .catch(() => false);
+};
+
+// eslint-disable-next-line no-unused-vars
+export const useCopyToClipboard = (): [boolean, (text: string) => void] => {
+  const [copied, setCopied] = useState(false);
+
+  const copyText = (text: string) => copyToClipboard(text).then(setCopied);
+
+  useEffect(() => {
+    if (!copied) {
```

Comment:
Change the logic to:

```jsx
if (copied) {
  const timerId = setTimeout(() => setCopied(false), 3000);

  return () => clearTimeout(timerId);
}

return undefined;
```

Reply:
An early return sounds cleaner to me

Reply:
Negated conditions are usually a bad practice if you can do the same condition without a negated operator. It's usually easier to read.

We often do negated conditions in JSX, because well, adding the `return null` after a gigantic block of JSX hinders the visibility of that statement.